lib/src/notification.dart

```diff
@@ -0,0 +1,389 @@
+    payload: data['talkjs'],
+  );
+}
+
+void _onSelectNotification(NotificationResponse details) {
+  final payload = details.payload;
+
+  print('📘 _onSelectNotification: $payload');
+
+  if (payload != null) {
+    final Map<String, dynamic> talkjsData = json.decode(payload);
+    final String notificationId = talkjsData['conversation']['id'];
+    _activeNotifications.remove(notificationId);
+  }
+}
+
+void _onFCMTokenRefresh(String token) {
+  print('📘 Firebase onTokenRefresh: $token');
+
+  fcmToken = token;
+
+  // TODO: Update the token on the Talkjs server once we have the data layer SDK ready
+}
+
+Future<void> registerAndroidPushNotificationHandlers(AndroidChannel androidChannel) async {
+  // Get the token each time the application loads
+  fcmToken = await FirebaseMessaging.instance.getToken();
+  print('📘 Firebase token: $fcmToken');
+
+  // Update the token each time it refreshes
+  FirebaseMessaging.instance.onTokenRefresh.listen(_onFCMTokenRefresh);
+
+  await _flutterLocalNotificationsPlugin.initialize(
+    InitializationSettings(
+      android: AndroidInitializationSettings('@mipmap/ic_launcher'),
+    ),
+    onDidReceiveNotificationResponse: _onSelectNotification,
+  );
+
+  _androidChannel = androidChannel;
+
+  try {
+    final activeNotifications = await _flutterLocalNotificationsPlugin
+      .resolvePlatformSpecificImplementation<AndroidFlutterLocalNotificationsPlugin>()
+      ?.getActiveNotifications();
+
+      if (activeNotifications != null) {
+        for (final displayedNotification in activeNotifications) {
+          if (displayedNotification.payload != null) {
+            final Map<String, dynamic> talkjsData = json.decode(displayedNotification.payload!);
+
+            if ((talkjsData['conversation'] != null) && (talkjsData['conversation']['id'] != null)) {
+              print('📘 Existing notification: ${displayedNotification.payload}');
+              final String notificationId = talkjsData['conversation']['id'];
+
+              if (!_showIdFromNotificationId.containsKey(notificationId)) {
+                _showIdFromNotificationId[notificationId] = _nextId;
+                _nextId += 1;
+              }
+
+              if (_activeNotifications[notificationId] == null) {
+                _activeNotifications[notificationId] = [displayedNotification.payload!];
+              } else {
+                _activeNotifications[notificationId]!.add(displayedNotification.payload!);
+              }
+            }
+          }
+        }
+      }
+  } on PlatformException {
+    // PlatformException is raised on Android < 6.0
+    // Simply ignoring this part
+  }
+
+  IsolateNameServer.registerPortWithName(_receivePort.sendPort, 'talkjsFCMPort');
+  _receivePort.listen((message) async => await _onReceiveMessageFromPort(message));
+
+  FirebaseMessaging.onBackgroundMessage(_onFCMBackgroundMessage);
+}
+
+Future<void> _onPush(String name, ApnsRemoteMessage message) async {
```

Comment:
it's in the planning stage at the moment.
Moreover, with the libraries I'm using it would be possible only on Android, as the flutter_apns package doesn't expose any handler on when the notification is clicked